### PR TITLE
Fix option race causing crashes

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Fix race where we are raising option changed events after a user has unsubscribed. This bug existed for awhile, but was made worse when we started raising a bunch of events when events didn't actually change. This fixes both issues.

*Review:* @dotnet/roslyn-ide

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@dotnet-bot, test vsi please.

File: src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs [hunk 1]
```diff
@@ -65,17 +65,16 @@ public void OnWorkspaceDisposed(Workspace workspace)
 
             private void OnGlobalOptionServiceOptionChanged(object sender, OptionChangedEventArgs e)
             {
-                var eventHandlers = GetEventHandlers();
-                if (eventHandlers.Length > 0)
+                _taskQueue?.ScheduleTask(() =>
```
Comment (tmeschter) [inline src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs]: Under what circumstances could `_taskQueue` be null?

Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs]: Tests, I think.

Comment (tmeschter) [inline src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs]: Ah, ok.

**Pull request merged**